chainstate/test-suite/src/tests/nft_issuance.rs

```diff
@@ -0,0 +1,1139 @@
+    })
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn nft_issuance_empty_name(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        let mut tf = TestFramework::default();
+        let outpoint_source_id = TestBlockInfo::from_genesis(&tf.genesis()).txns[0].0.clone();
+        let mut rng = make_seedable_rng(seed);
+
+        // Name is empty
+        let max_desc_len = tf.chainstate.get_chain_config().token_max_description_len();
+        let max_ticker_len = tf.chainstate.get_chain_config().token_max_ticker_len();
+        let result = tf
+            .make_block_builder()
+            .add_transaction(
+                TransactionBuilder::new()
+                    .add_input(
+                        TxInput::new(outpoint_source_id, 0),
+                        InputWitness::NoSignature(None),
+                    )
+                    .add_output(TxOutput::new(
+                        OutputValue::Token(TokenData::NftIssuanceV1(NftIssuanceV1 {
+                            metadata: Metadata {
+                                creator: random_creator(),
+                                name: vec![],
+                                description: random_string(&mut rng, 1..max_desc_len).into_bytes(),
+                                ticker: random_string(&mut rng, 1..max_ticker_len).into_bytes(),
+                                icon_uri: None,
+                                additional_metadata_uri: None,
+                                media_uri: None,
+                                media_hash: vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 0],
+                            },
+                        })),
+                        OutputPurpose::Transfer(Destination::AnyoneCanSpend),
+                    ))
+                    .build(),
+            )
+            .build_and_process();
+
+        assert!(matches!(
+            result,
+            Err(ChainstateError::ProcessBlockError(
+                BlockError::CheckBlockFailed(CheckBlockError::CheckTransactionFailed(
+                    CheckBlockTransactionsError::TokensError(
+                        TokensError::IssueErrorInvalidNameLength(_, _)
+                    )
+                ))
+            ))
+        ));
+    })
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn nft_issuance_invalid_name(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        let mut tf = TestFramework::default();
+        let outpoint_source_id = TestBlockInfo::from_genesis(&tf.genesis()).txns[0].0.clone();
+        let mut rng = make_seedable_rng(seed);
+
+        let max_desc_len = tf.chainstate.get_chain_config().token_max_description_len();
+        let max_name_len = tf.chainstate.get_chain_config().token_max_name_len();
+        let max_ticker_len = tf.chainstate.get_chain_config().token_max_ticker_len();
+
+        // try all possible chars for ticker and ensure everything fails except for alphanumeric chars
+        for c in u8::MIN..u8::MAX {
```

Comment:
I think `proptest` is ideal for this purpose. You just need to feed a regexp and it fill generate tickers and test the whole range.

The only possible problem would be how to init `make_seedable_rng`, but maybe we don't need it at all if we let proptest generate all the strings